tests/beta/test_default_qubit_tf.py

```diff
@@ -1045,174 +1020,195 @@ def circuit(a, b):
-    def test_tf_interface_classical_diff(self, cost_with_decomposition, interface, tol):
-        """Tests that the gradient of an arbitrary U3 gate (that gets
-        decomposed) is correct using the TensorFlow interface and the classical
-        diff method"""
-        # TODO: once the decomposition of operations and the PassThruQNode are
-        # compatible, merge this case into the previous one
-        import tensorflow as tf
+class TestSamplesNonAnalytic:
+    """Tests for sampling and non-analytic mode"""
 
-        params = tf.Variable(self.p, dtype=tf.float64)
+    def test_sample_observables(self):
+        """Test that the device allows for sampling from observables."""
+        shots = 100
+        dev = qml.device("default.qubit.tf", wires=2, shots=shots)
+
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a):
+            qml.RX(a, wires=0)
+            return qml.sample(qml.PauliZ(0))
+
+        a = tf.Variable(0.54)
+        res = circuit(a)
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (1, shots)
+        assert set(res[0].numpy()) == {-1, 1}
+
+    def test_sample_observables_non_differentiable(self):
+        """Test that sampled observables cannot be differentiated."""
+        shots = 100
+        dev = qml.device("default.qubit.tf", wires=2, shots=shots)
+
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a):
+            qml.RX(a, wires=0)
+            return qml.sample(qml.PauliZ(0))
+
+        a = tf.Variable(0.54)
 
         with tf.GradientTape() as tape:
-            tape.watch(params)
-            res = cost_with_decomposition(params)
+            res = circuit(a)
 
-        assert np.allclose(res.numpy(), self.expected_cost, atol=tol, rtol=0)
+        assert tape.gradient(res, a) is None
 
-        res = tape.gradient(res, params)
-        assert np.allclose(res.numpy(), self.expected_grad, atol=tol, rtol=0)
+    def test_estimating_probability(self, tol):
+        """Test that the probability is accurately estimated."""
+        dev = qml.device("default.qubit.tf", wires=2, analytic=False, shots=1000)
 
-    def test_error_classical_diff_torch(self, torch_support, tol):
-        """Tests that an error is raised if for the classical differentiation
-        method when using the Torch interface"""
-        if not torch_support:
-            pytest.skip("Skipped, no torch support")
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit():
+            qml.PauliX(0)
+            return qml.probs(wires=[0])
 
-        import torch
-        from torch.autograd import Variable
+        res = circuit()
 
-        interface = "torch"
-        diff_method = "backprop"
+        assert isinstance(res, tf.Tensor)
 
-        params = Variable(torch.tensor(self.p), requires_grad=True)
+        expected = np.array([0, 1])
+        assert np.allclose(res, expected, atol=tol, rtol=0)

```

Comment:
Can we have a test case here where `qml.probs` uses more than one wire?

Reply:
Done :+1: